pkg/metricsadapter/provider/resourcemetrics.go

```diff
@@ -150,10 +148,19 @@ func (r *ResourceMetricsProvider) queryPodMetricsByName(name, namespace string)
 		_, err := sci.Lister(PodsGVR).ByNamespace(namespace).Get(name)
 		return err
 	}
-	metricsQueryFunc := func(sci genericmanager.SingleClusterInformerManager, _ string) (interface{}, error) {
-		metrics, err := sci.GetClient().Resource(podMetricsGVR).
+	metricsQueryFunc := func(sci genericmanager.SingleClusterInformerManager, clusterName string) (interface{}, error) {
+		metricsValue, err := sci.GetClient().Resource(podMetricsGVR).
 			Namespace(namespace).Get(context.Background(), name, metav1.GetOptions{})
-		return metrics, err
+		if err != nil {
+			return nil, err
+		}
+		annotations := metricsValue.GetAnnotations()
```

Comment:
can we extract this as a function? it has duplicated code

Reply:
Updated.